docs/design-v2.md

```diff
@@ -1,133 +1,77 @@
-The `ControllerPublishVolume` request is complete once the `AzVolumeAttachment` instance for 
-primary node has been crerated. As each the attachment request completes, the controller sets 
-the `.Status.State` field to `Attached` and adds a finalizer to each `AzVolumeAttachment` 
-instance. When the `NodeStageVolume` API is called, the CSI Node plug-in will wait for the its
-`AzVolumeAttachment` instance's state to reach `Attached` before staging the mount point to 
-the disk.
-
-When the `ControllerUnpublishVolume` API in the CSI Controller plug-in is called, it deletes 
-the `AzVolumeAttachment` instance for the primary node. The controller responds by detaching 
-the managed disk from the primary node. It removes the finalizer from the `AzVolumeAttachment`
-instance when the detach operation completes. The `ControllerUnpublishVolume` request is 
-complete when the detach operation has completed and corresponding `AzVolumeAttachment` has 
-been removed from the object store.
+The `AzVolumeAttachment` custom resource represents the attachment of a managed disk to a specific node. The controller for this custom resource runs in the controller plug-in and watches for changes in the `AzVolumeAttachment` instances.
+
+An `AzVolumeAttachment` instance representing the primary node attachment is created by the `ControllerPublishVolume` API in the CSI Controller plug-in. If an instance for the current node already exists, it is updated to represent the primary node attachment. The `AzVolumeAttachment` controller then attaches the manage disk to the primary node (downgrading to attachment replica or detaching any previous primary attachment if one exists). It then creates additional `AzVolumeAttachment` instances representing attachment replicas and attaches the shared manage disk to a number of backup nodes as specified by the `maxMountReplicaCount` parameter in the `StorageClass` instance of the `PersistentVolumeClaim` of the managed disk. The `ControllerPublishVolume` request is complete once the `AzVolumeAttachment` instance for primary node has been crerated. As each the attachment request completes, the controller sets the `.Status.State` field to `Attached` and adds a finalizer to each `AzVolumeAttachment` instance. When the `NodeStageVolume` API is called, the CSI Node plug-in will wait for the its
+`AzVolumeAttachment` instance's state to reach `Attached` before staging the mount point to the disk.
+
+When the `ControllerUnpublishVolume` API in the CSI Controller plug-in is called, it deletes the `AzVolumeAttachment` instance for the primary node. The controller responds by detaching the managed disk from the primary node. It removes the finalizer from the `AzVolumeAttachment` instance when the detach operation completes. The `ControllerUnpublishVolume` request is complete when the detach operation has completed and corresponding `AzVolumeAttachment` has been removed from the object store.
```

Comment:
And the controller schedules a garbage collection to delete replica attachments if no subsequent `ControllerPublishVolume` request is made for the volume after its detachment from the primary node within 5 minutes.

Reply:
Done.